bsip-0048.md

```diff
@@ -10,45 +10,61 @@
 
 # Abstract
 
-The possibility of creating assets on the BitShares Blockchain is a core feature
-of our platform. In the case of an ico, the ico-holders can create an asset with
-a fixed maximum supply. However this maximum supply can be changed by the
-asset creator (issuer) anytime. This exposes participators ico to the risk,
-of loosing value of their bought asset.
+The possibility of creating assets on the BitShares Blockchain is a core feature of our platform. However some things are too maleable when they should not be. Other things are inflexible when they should be.
 
-With this BSIP a flag will be introduced which forbids the modification of the maximum
-supply of an issuer created asset.
+In the case of an ICO, the ICO holders can create an asset with a maximum supply. However this maximum supply can be changed by the asset creator (issuer) anytime. This exposes participants in the ICO to the risk of losing value of their purchased asset.
+
+With this BSIP a flag will be introduced which forbids the modification of the maximum supply of an issuer created asset.
+
+In addition, another new flag will be introduced that prevents any more tokens from being issued in the future.
+
+And finally, the ability to change the precision of an asset will now be available as long as current supply is zero.
 
 # Motivation
 
-The implementation of a flag, to disable the modification of the maximum supply for 
-issuer created asset, has effects for both sides. On the one side the security for the
-participators or non-issuer users increases, because the value of their assets (tokens)
-can't be manipulated by exposing the market to more created tokens. And on the other side
-it is good marketing for the ico-hoders, because they can advertise with their fixed
-maximum supply.
+A flag to disable the modification of the maximum supply for issuer created asset will be implemented. This has benefits for both the participants and the ICO holder. Participants in the ICO are assured the value of their assets (tokens) will not be diluted by issuers creating more tokens. ICO holders are able to market the fact that the maximum supply is fixed.
+
+A flag to disable issuing allows token holders to be assured that the current supply will remain fixed, even if there are unissued tokens.
+
+The ability to change the precision of a created asset adds flexibility in the asset creation process.
 
 # Rational
 
-The `disable_modify_max_supply` is a new created flag in the `asset_issuer_permissions`.
-The flag can be activated to forbid the modification of the `max_supply` (`asset_object.options`).
-Once the flag is activated, it can't be deactivated.
+The `disable_modify_max_supply` is a newly created flag in the `asset_issuer_permissions`. The flag can be activated to forbid the modification of the `max_supply` (`asset_object.options`). Once the flag is set to `true`, it can only be set to `false` if `current_supply` is zero.
```

Comment:
Please be more specific. "flags" are different from "permissions". "permission" implies the permission to change the flag value, whereas the "flag" implies a specific restriction.
Same for "disable_issue".

Reply:
Let me see if I understand you correctly. From the programmer standpoint, disable_modify_max_supply is a flag, as it is a bit. But that muddies the waters, as others see it as a "permission". So I should re-word the paragraph to make `disable_modify_max_supply`  to be a "permission", as it is part of the "asset_issuer_permissions". 

If the above is correct, I will modify the paragraph below as well.

Reply:
What I mean is this: https://github.com/bitshares/bitshares-core/blob/master/libraries/protocol/include/graphene/protocol/asset_ops.hpp#L56-L59
Assets have *both* flags and permissions. The `flag` tells what is currently allowed. The `permission` tells if the flag setting can be changed.
An inactive `permission` can only be enabled when `supply == 0`.

Reply:
The lightbulb just went on. Thanks for flipping the switch.

Reply:
Is this considered resolved?

Reply:
It's partially resolved. In the last commit one sentence of two got updated.